components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -311,23 +312,76 @@ public void deleteUser(String userId) throws NotFoundException, CharonException
-        ClaimMapping[] userClaims;
-        ClaimMapping[] extensionClaims = null;
         List<Object> users = new ArrayList<>();
-        //0th index is to store total number of results;
+        //0th index is to store total number of results.
         users.add(0);
+
+        String[] userNames;
+
+        if (StringUtils.isNotEmpty(domainName)) { // When domain is not null or not empty.
+            userNames = getAllUsernamesByDomainName(domainName);
+        } else { // When domain is null or empty.
+            userNames = getAllUsernamesByAllDomainNames();
+        }
+
+        if (userNames != null) {
+            users.set(0, userNames.length); // Set total number of results to 0th index.
+            users.addAll(getUserDetails(userNames, requiredAttributes)); // Set user details from index 1.
+        }
+
+        return users;
+    }
+
+    private String[] getAllUsernamesByDomainName(String domainName) throws CharonException {
+
+        String[] userNames = null;
+        UserStoreManager tempCarbonUM = carbonUM.getSecondaryUserStoreManager(domainName);
+
         try {
             Map<String, String> scimToLocalClaimsMap = SCIMCommonUtils.getSCIMtoLocalMappings();
-            String userIdLocalClaim = scimToLocalClaimsMap.get(SCIMConstants
-                    .CommonSchemaConstants.ID_URI);
-            String[] userNames = null;
+            String userIdLocalClaim = scimToLocalClaimsMap.get(SCIMConstants.CommonSchemaConstants.ID_URI);
+            if (StringUtils.isNotBlank(userIdLocalClaim)) {
+                userNames = tempCarbonUM.getUserList(userIdLocalClaim, "*", null);
+            }
+
+            return userNames;
+        } catch (UserStoreException e) {
+            throw new CharonException(String.format("Error while listing users from domain: %s.", domainName), e);
+        }
+    }
+
+    private String[] getAllUsernamesByAllDomainNames() throws CharonException {
+
+        String[] userNames = null;
+
+        try {
+            Map<String, String> scimToLocalClaimsMap = SCIMCommonUtils.getSCIMtoLocalMappings();
+            String userIdLocalClaim = scimToLocalClaimsMap.get(SCIMConstants.CommonSchemaConstants.ID_URI);
             if (StringUtils.isNotBlank(userIdLocalClaim)) {
                 userNames = carbonUM.getUserList(userIdLocalClaim, "*", null);
             }
-            if (userNames != null && userNames.length != 0) {
+
+            return userNames;
+        } catch (UserStoreException e) {
+            throw new CharonException("Error while listing users across all domains. ", e);
+        }
+    }
+
+    private List<Object> getUserDetails(String[] userNames, Map<String, Boolean> requiredAttributes)
+            throws CharonException {
+
+        List<Object> users = new ArrayList<>();
+
+        if (ArrayUtils.isEmpty(userNames)) {
+            if (log.isDebugEnabled()) {
+                log.debug("Users for this filter does not exist in the system.");
```

Comment:
Update the debug log message in a reasonable way. Here we don't filter users so better to rephrase it

Reply:
Do we have separate "getUserDetails()" method for filter users?

Reply:
"getFilteredUserDetails()" is a separate method for get user details of filtered users. "getUserDetails()" method for get user details of list users.